main.js

```diff
@@ -1,6 +1,64 @@
-import './style.css'
-
-document.querySelector('#app').innerHTML = `
-  <h1>Hello Vite!</h1>
-  <a href="https://vitejs.dev/guide/features.html" target="_blank">Documentation</a>
-`
+const api_url = "https://api.themoviedb.org/3/discover/movie?sort_by=popularity.desc&api_key=04c35731a5ee918f014970082a0088b1&page=1";
+const img_path = "https://image.tmdb.org/t/p/w1280";
+const search_api = "https://api.themoviedb.org/3/search/movie?&api_key=04c35731a5ee918f014970082a0088b1&query=";
+const movieGrid = document.querySelector("#movie_grid")
+const getMovies = async (api) => {
+    const response = await fetch(api)
+    const data = await response.json()
+    let display_size = data.results.slice(0,15)
+    let movie_grid = document.getElementById("movie_grid");
+    console.log(movie_grid)
+    let no_results = document.getElementById("no_results_wrapper");
+    if (display_size.length) {
+        movie_grid.style.display = "flex";
+        no_results.style.display = "none";
+        showMovies(display_size);
+      } else {
+        movie_grid.style.display = "none";
+        no_results.style.display = "block";
+      }
+}
+const showMovies = (data) => {
+    movieGrid.textContent = "";
+    data.forEach(
+        (item) => {
+            const movieGridItem = document.createElement("div")
+            movieGridItem.classList.add("movie-grid-item")
+            let movieGridInnerTag = document.createElement("div")
+            movieGridInnerTag.classList.add("movie-grid-inner")
+            let imgTag = document.createElement("img")
+            imgTag.src = img_path + item.poster_path 
+            imgTag.alt = "banner-image"
+            let movieContentTag = document.createElement("div")
+            movieContentTag.classList.add("movie-content")
+            let h3Tag = document.createElement("h3")
+            h3Tag.classList.add("h3-tag")
+            let anchorTag = document.createElement("a")
+            anchorTag.classList.add("book-now-btn")
+            anchorTag.title = "book-now-btn"
+            anchorTag.id = "book-now-btn"
```

Comment:
do not use id for a html element that will be present more than once in the document. id is considered to be unique for single page it will show warning in the console if there are multiple elements with same ids.

Reply:
Noted and removed.